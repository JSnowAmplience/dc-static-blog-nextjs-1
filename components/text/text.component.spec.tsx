import renderer from 'react-test-renderer';
import Text from './text.component';
import AmplienceText from '../../common/interfaces/text.interface';

describe('Text', (): void => {
  it('should render some text as html', (): void => {
    const text: AmplienceText = {
      text:
        '<h1 id="this-is-my-first-bloblog">This is my first Bloblog</h1>\n<p>This is some sample text.</p>\n<h2 id="look-a-header">Look-a-header</h2>\n<p>This is some more sample text</p>\n'
    };

    const wrapper = renderer.create(<Text text={text.text} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
