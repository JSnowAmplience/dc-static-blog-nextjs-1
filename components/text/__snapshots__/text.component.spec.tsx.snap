// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render some text as html 1`] = `
<div
  dangerouslySetInnerHTML={
    Object {
      "__html": "<h1 id=\\"this-is-my-first-bloblog\\">This is my first Bloblog</h1>
<p>This is some sample text.</p>
<h2 id=\\"look-a-header\\">Look-a-header</h2>
<p>This is some more sample text</p>
",
    }
  }
/>
`;
